### PR TITLE
feat: Proxy Server Construct

### LINKED ISSUE #99: feat: Construct to create a proxy
### Description

A construct that will allow you to stand up a proxy server either in a VPC you provide of one that is created for you.

### Use Case

Currently many enterprises utilize proxies.  A construct that creates a proxy would allow developers to test thier code against a proxy when it is not available through the customer.

### Proposed Solution

Develop a construct that will stand up a proxy server for you. This should be able to be deployed either into a VPC you provide or if one is not provided create on for you. Additionally developers should be able to provide a list of domains that are allowed by the proxy. 

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by BlakeRBarnett**
Fixes #99

### CODE REVIEW

File: resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/group_vars/REPLACE_ME_all.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (BlakeRBarnett) [inline resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/group_vars/REPLACE_ME_all.yml:1]: Decide which file to use. I don't think I need the REPLACE_ME_* files anymore.

File: resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/roles/ansible-role-cloudwatch/defaults/REPLACE_ME_main.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (BlakeRBarnett) [inline resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/roles/ansible-role-cloudwatch/defaults/REPLACE_ME_main.yaml:1]: Decide which file to use. I don't think I need the REPLACE_ME_* files anymore.

File: resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/roles/ansible-role-squid/defaults/REPLACE_ME_main.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (BlakeRBarnett) [inline resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/roles/ansible-role-squid/defaults/REPLACE_ME_main.yaml:1]: Decide which file to use. I don't think I need the REPLACE_ME_* files anymore.

File: resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/roles/ansible-role-squid/templates/REPLACE_ME_squid.conf [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# Recommended minimum configuration:
```
Comment (BlakeRBarnett) [inline resources/constructs/proxyServer/ansibleResources/s3Deployment/ansible-playbook-proxy/roles/ansible-role-squid/templates/REPLACE_ME_squid.conf:1]: Decide which file to use. I don't think I need the REPLACE_ME_* files anymore.

File: resources/constructs/proxyServer/ansibleResources/s3Deployment/squid-resources/cf_resources/REPLACE_ME_squid.conf [hunk 5]
```diff
@@ -0,0 +1,82 @@
+# Recommended minimum configuration:
```
Comment (BlakeRBarnett) [inline resources/constructs/proxyServer/ansibleResources/s3Deployment/squid-resources/cf_resources/REPLACE_ME_squid.conf:1]: Decide which file to use. I don't think I need the REPLACE_ME_* files anymore.

File: src/constructs/proxyServer/integ.proxyServer.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (BlakeRBarnett) [inline src/constructs/proxyServer/integ.proxyServer.ts:1]: This file can be removed from here and inserted from integ test when need inorder to produce a .js file for testing

File: resources/constructs/proxyServer/ansibleResources/s3Deployment/squid-resources/cf_resources/approved-domains.conf [hunk 7]
```diff
@@ -0,0 +1 @@
+.amazon.com
```
Comment (BlakeRBarnett) [inline resources/constructs/proxyServer/ansibleResources/s3Deployment/squid-resources/cf_resources/approved-domains.conf:1]: The contents of this file will be generated based on user inputs to the construct

File: src/constructs/proxyServer/createAnsiblePlaybook.ts [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable security/detect-object-injection */
+/* eslint-disable security/detect-non-literal-fs-filename */
+//import * as fs from 'fs';
+import * as path from 'path';
+import * as cdk from 'aws-cdk-lib';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as s3Deploy from 'aws-cdk-lib/aws-s3-deployment';
+import { Construct } from 'constructs';
+//import * as util from 'util';
```
Comment (BlakeRBarnett) [inline src/constructs/proxyServer/createAnsiblePlaybook.ts:11]: Remove

File: src/constructs/proxyServer/createAnsiblePlaybook.ts [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+//import * as stream from 'stream'
```
Comment (BlakeRBarnett) [inline src/constructs/proxyServer/createAnsiblePlaybook.ts:12]: Remove

File: src/constructs/proxyServer/createAnsiblePlaybook.ts [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+ * Inputs: approvedDomains - array of approved domains as strings
+ *         filepath - file to write the approvedDomains to
+ */
+async function updateApprovedDomains(
+  approvedDomains: string[],
+  filepath: string
+) {
+  const fs = require('fs');
+  const writeStream = fs.createWriteStream(filepath);
+  //const writeStream = fs.createWriteStream(filepath);
```
Comment (BlakeRBarnett) [inline src/constructs/proxyServer/createAnsiblePlaybook.ts:65]: Remove

File: src/constructs/proxyServer/createAnsiblePlaybook.ts [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+  //const pathName = writeStream.path;
```
Comment (BlakeRBarnett) [inline src/constructs/proxyServer/createAnsiblePlaybook.ts:66]: Remove

File: src/constructs/proxyServer/proxyServer.ts [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+ * Creates a Proxy construct. This construct allows you to deploy a proxy server
+ * in an Auto Scaling Group fronted by a Networl Loadbalancer
+ *
+ * @param scope The parent creating construct (usually `this`).
+ * @param id The construct's name.
+ * @param this A `proxyProps` interface.
+ */
+
+export class Proxy extends Construct {
+  //public readonly network: proxyNetwork.ProxyNetwork;
```
Comment (BlakeRBarnett) [inline src/constructs/proxyServer/proxyServer.ts:51]: Remove

File: src/constructs/proxyServer/proxyResources.ts [hunk 13]
```diff
@@ -0,0 +1,552 @@
...
+      path: '/',
+      managedPolicies: [
+        iam.ManagedPolicy.fromAwsManagedPolicyName(
+          'AmazonSSMManagedInstanceCore'
+        ),
+        iam.ManagedPolicy.fromAwsManagedPolicyName(
+          'CloudWatchAgentServerPolicy'
+        ),
+      ],
+      //inlinePolicies: { proxyPolicy: ProxyPolicy }
```
Comment (BlakeRBarnett) [inline src/constructs/proxyServer/proxyResources.ts:62]: Remove

**Comment by maafk** [comment]
@BlakeRBarnett Can you address the build failures?

**Comment by BlakeRBarnett** [comment]
> 

Yes sir I will be working on this over the weekend

**Pull request closed without merging**